source/extensions/omni.isaac.lab/test/sensors/test_frame_transformer.py

```diff
@@ -302,6 +322,87 @@ def test_frame_transformer_feet_wrt_thigh(self):
+        for count in range(100):
+            # # reset
+            if count % 25 == 0:
+                # reset root state
+                root_state = scene.articulations["robot"].data.default_root_state.clone()
+                root_state[:, :3] += scene.env_origins
+                joint_pos = scene.articulations["robot"].data.default_joint_pos
+                joint_vel = scene.articulations["robot"].data.default_joint_vel
+                # -- set root state
+                # -- robot
+                scene.articulations["robot"].write_root_state_to_sim(root_state)
+                scene.articulations["robot"].write_joint_state_to_sim(joint_pos, joint_vel)
+                # reset buffers
+                scene.reset()
+
+            # set joint targets
+            robot_actions = default_actions + 0.5 * torch.randn_like(default_actions)
+            scene.articulations["robot"].set_joint_position_target(robot_actions)
+            # write data to sim
+            scene.write_data_to_sim()
+            # perform step
+            self.sim.step()
+            # read data from sim
+            scene.update(sim_dt)
+
+            # check absolute frame transforms in world frame
+            # -- ground-truth
+            root_pose_w = scene.articulations["robot"].data.root_state_w[:, :7]
+            cube_pos_w_gt = scene.rigid_objects["cube"].data.root_state_w[:, :3]
+            cube_quat_w_gt = scene.rigid_objects["cube"].data.root_state_w[:, 3:7]
+            # -- frame transformer
+            source_pos_w_tf = scene.sensors["frame_transformer"].data.source_pos_w
+            source_quat_w_tf = scene.sensors["frame_transformer"].data.source_quat_w
+            cube_pos_w_tf = scene.sensors["frame_transformer"].data.target_pos_w.squeeze()
+            cube_quat_w_tf = scene.sensors["frame_transformer"].data.target_quat_w.squeeze()
+
+            # check if they are same
+            torch.testing.assert_close(root_pose_w[:, :3], source_pos_w_tf, rtol=1e-3, atol=1e-3)
+            torch.testing.assert_close(root_pose_w[:, 3:], source_quat_w_tf, rtol=1e-3, atol=1e-3)
+            torch.testing.assert_close(cube_pos_w_gt, cube_pos_w_tf, rtol=1e-3, atol=1e-3)
+            torch.testing.assert_close(cube_quat_w_gt, cube_quat_w_tf, rtol=1e-3, atol=1e-3)
+
+            # check if relative transforms are same
+            cube_pos_source_tf = scene.sensors["frame_transformer"].data.target_pos_source
+            cube_quat_source_tf = scene.sensors["frame_transformer"].data.target_quat_source
+            # ground-truth
+            cube_pos_b, cube_quat_b = math_utils.subtract_frame_transforms(
+                root_pose_w[:, :3], root_pose_w[:, 3:], cube_pos_w_tf, cube_quat_w_tf
+            )
+            # check if they are same
+            torch.testing.assert_close(cube_pos_source_tf[:, 0], cube_pos_b, rtol=1e-3, atol=1e-3)
+            torch.testing.assert_close(cube_quat_source_tf[:, 0], cube_quat_b, rtol=1e-3, atol=1e-3)
```

Comment:
Why are these tolerances so high? They should be exactly the same, right?

Reply:
I removed them, and the tests still work. Please be careful of atols and rtols. We had them in some other places, and a bug went unidentified for Cameras.